plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -900,7 +900,8 @@ def main():
         proxmox = ProxmoxAPI(api_host, user=api_user, password=api_password, verify_ssl=validate_certs)
         global VZ_TYPE
         global PVE_MAJOR_VERSION
-        PVE_MAJOR_VERSION = 3 if proxmox_version(proxmox) < LooseVersion('4.0') else 4
+        pve_version = re.sub(r'-\d$', '', proxmox.version.get()['version'])
+        PVE_MAJOR_VERSION = 3 if float(pve_version) < 4.0 else int(pve_version.split('.')[0])
```

Comment:
Why are you using hand-written version number parsing instead of the helping methods we already have and `LooseVersion`? That doesn't seem like a good idea to me. I think you should call `proxmox_version(proxmox)` and work with its result.

Reply:
Okay, sounds legit ;) I've changed the single line where the version get's actually compared to work with the LooseVersion object. Much simpler.